src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -582,18 +582,29 @@ public void Dispose()
             return context.ResolveSatelliteAssembly(assemblyName);
         }
 
-        private Assembly? GetFirstResolvedAssembly(AssemblyName assemblyName)
+        private Assembly? GetFirstResolvedAssemblyFromResolvingEvent(AssemblyName assemblyName)
         {
             Assembly? resolvedAssembly = null;
 
-            Func<AssemblyLoadContext, AssemblyName, Assembly>? assemblyResolveHandler = _resolving;
+            Func<AssemblyLoadContext, AssemblyName, Assembly>? resolvingHandler = _resolving;
 
-            if (assemblyResolveHandler != null)
+            if (resolvingHandler != null)
             {
                 // Loop through the event subscribers and return the first non-null Assembly instance
-                foreach (Func<AssemblyLoadContext, AssemblyName, Assembly> handler in assemblyResolveHandler.GetInvocationList())
+                foreach (Func<AssemblyLoadContext, AssemblyName, Assembly> handler in resolvingHandler.GetInvocationList())
                 {
                     resolvedAssembly = handler(this, assemblyName);
```

Comment:
Not necessarily for this change: What about exceptions - I wonder what is the tracing experience when the handler throws. I know it ends the entire bind operation - I'm just wondering if the reported trace/exception gives enough data.

The exception (and the trace for the exception) will contain the callstack - so that points to the handler, but we probably loose the context information (assembly name, ALC, ...) - that said maybe it's easy to reconstruct those from the bind start/end traces.

Can we please add this to the list of "Things to validate and potentially improve"?

Reply:
Ah, good point.

I think it can get particularly confusing since if the `Resolving` handlers throw a `FileNotFoundException`, we will still call the `AssemblyResolve` handlers (but not for other exceptions), so it could still end in a successful load.

I was thinking we'd want tracing to have the resolution attempted events with a failure result / error message for that stage - will definitely make sure we validate what it looks like and track desired improvements.